ui/hooks/useMultichainAccountTotalFiatBalance.ts

```diff
@@ -0,0 +1,107 @@
+import { useSelector } from 'react-redux';
+import { InternalAccount, isEvmAccountType } from '@metamask/keyring-api';
+import {
+  getMultichainCurrencyImage,
+  getMultichainBalances,
+  getMultichainCoinRates,
+  getMultichainNetwork,
+} from '../selectors/multichain';
+import { formatCurrency } from '../helpers/utils/confirm-tx.util';
+import { MULTICHAIN_NATIVE_CURRENCY_TO_CAIP19 } from '../../shared/constants/multichain/networks';
+import { getTokenFiatAmount } from '../helpers/utils/token-util';
+import { useMultichainSelector } from './useMultichainSelector';
+
+export const EMPTY_VALUES = {
+  formattedFiat: '0',
+  totalFiatBalance: '0',
+  tokensWithBalances: [],
+  loading: false,
+  orderedTokenList: [],
+  error: null,
+};
+
+export const useMultichainAccountTotalFiatBalance = (
+  account: InternalAccount,
+): {
+  formattedFiat: string;
+  totalFiatBalance: string;
+  tokensWithBalances: {
+    address: string;
+    symbol: string;
+    decimals: string;
+    isERC721: boolean;
+    image: string;
+  }[];
+  loading: boolean;
+  orderedTokenList: { iconUrl: string; symbol: string; fiatBalance: string }[];
+  error: string | null;
+} => {
+  if (isEvmAccountType(account.type)) {
+    return EMPTY_VALUES;
+  }
+
+  // The fiat denomination to display
+  // TODO: fix me when we have a know how to deal with non usd currencies.
+  const currentCurrency = 'usd';
+  const { network } = useMultichainSelector(getMultichainNetwork, account);
+  const ticker = isEvmAccountType(account.type) ? 'ETH' : 'BTC';
+
+  // get rate from rate controller
+  const multichainCoinRates = useSelector(getMultichainCoinRates);
+  // rates uses the lower case ticker
+  const { conversionRate } = multichainCoinRates[network.ticker.toLowerCase()];
```

Comment:
Maybe re-import the `getMultichainConversionRate` from my PR here: https://github.com/MetaMask/metamask-extension/pull/25200/files#diff-60c97f66ba4c1d03dd112bd4c20a71f1dd5f20203995374487c2b91478132c83

Reply:
Makes sense